train_management/views/traintimetabletemplate.py

```diff
@@ -38,3 +39,29 @@ class TrainTimetableTemplateDeleteView(DvzoDeleteView):
     model = TrainTimetableTemplate
     template_name = "train_management/confirm_delete.html"
     success_url = reverse_lazy("train-timetable-template-list")
+
+
+class CreateTimetablesFromTemplateView(DvzoView):
+    permission_required = 'train_management_add_traintimetable'
+
+    def post(self, request, **kwargs):
+        timetable_ids = request.POST["timetables"].split(",")
+        train_id = request.POST["pk"]
+        timetables = [get_object_or_404(TrainTimetableTemplate, pk=pk) for pk in timetable_ids]
```

Comment:
Must all timetables be "valid" or should it process valid ones but message back that some were not added for some reason if any fails?

Reply:
Templates must be valid. Since templates have the same fields (and required fields) this should not be a problem